### ISSUE TITLE
FTL crashed!

### ISSUE THREAD

**Issue by libertyx82** [pi-hole/FTL#1005]
### Versions
Pi-hole version is v5.2.2 (Latest: v5.2.2)
AdminLTE version is v5.2.2 (Latest: v5.2.2)
FTL version is v5.3.4 (Latest: v5.3.4)

### Platform
Debian 10, Debian Buster

```
[2020-12-26 05:03:30.582 27965/F619] Resizing "FTL-per-client-regex" from 4096 to (187 * 22) == 4114 (/dev/shm: 1.9MB used, 1.0GB total)
[2020-12-26 05:03:30.641 27966/F619] Resizing "FTL-per-client-regex" from 4096 to (188 * 22) == 4136 (/dev/shm: 1.9MB used, 1.0GB total)
[2020-12-26 05:03:43.953 27965/F619] ERROR: get_per_client_regex(187, 1): Out of bounds (4115 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 2): Out of bounds (4116 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 3): Out of bounds (4117 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 4): Out of bounds (4118 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 5): Out of bounds (4119 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 6): Out of bounds (4120 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 7): Out of bounds (4121 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 8): Out of bounds (4122 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 9): Out of bounds (4123 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 10): Out of bounds (4124 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 11): Out of bounds (4125 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 12): Out of bounds (4126 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 13): Out of bounds (4127 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 14): Out of bounds (4128 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 15): Out of bounds (4129 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 16): Out of bounds (4130 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 17): Out of bounds (4131 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 18): Out of bounds (4132 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 19): Out of bounds (4133 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 20): Out of bounds (4134 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:43.954 27965/F619] ERROR: get_per_client_regex(187, 21): Out of bounds (4135 > 188 * 22, shm_per_client_regex.size = 4114)!
[2020-12-26 05:03:44.015 27967/F619] !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
[2020-12-26 05:03:44.016 27967/F619] ---------------------------->  FTL crashed!  <----------------------------
[2020-12-26 05:03:44.016 27967/F619] !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
[2020-12-26 05:03:44.016 27967/F619] Please report a bug at https://github.com/pi-hole/FTL/issues
[2020-12-26 05:03:44.016 27967/F619] and include in your report already the following details:
[2020-12-26 05:03:44.016 27967/F619] FTL has been running for 669693 seconds
[2020-12-26 05:03:44.016 27967/F619] FTL branch: master
[2020-12-26 05:03:44.016 27967/F619] FTL version: v5.3.2
[2020-12-26 05:03:44.016 27967/F619] FTL commit: 0790cf7
[2020-12-26 05:03:44.016 27967/F619] FTL date: 2020-12-02 21:28:39 +0000
[2020-12-26 05:03:44.016 27967/F619] FTL user: started as pihole, ended as pihole
[2020-12-26 05:03:44.016 27967/F619] Compiled for x86_64 (compiled on CI) using gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
[2020-12-26 05:03:44.016 27967/F619] Process details: MID: 619
[2020-12-26 05:03:44.016 27967/F619]                  PID: 27967
[2020-12-26 05:03:44.016 27967/F619]                  TID: 27967
[2020-12-26 05:03:44.016 27967/F619]                  Name: pihole-FTL
[2020-12-26 05:03:44.016 27967/F619] Received signal: Segmentation fault
[2020-12-26 05:03:44.016 27967/F619]      at address: 0x7f1b40b2e000
[2020-12-26 05:03:44.016 27967/F619]      with code:  SEGV_MAPERR (Address not mapped to object)
[2020-12-26 05:03:44.016 27967/F619] Backtrace:
[2020-12-26 05:03:44.017 27967/F619] B[0000]: /usr/bin/pihole-FTL(+0x534e5) [0x56240d1a24e5]
[2020-12-26 05:03:44.019 27967/F619] L[0000]: N/A (0x534e5)
[2020-12-26 05:03:44.019 27967/F619] B[0001]: /lib/x86_64-linux-gnu/libpthread.so.0(+0x12730) [0x7f1b4098f730]
[2020-12-26 05:03:44.019 27967/F619] B[0002]: /usr/bin/pihole-FTL(get_per_client_regex+0x33) [0x56240d1a1ec3]
[2020-12-26 05:03:44.020 27967/F619] L[0002]: N/A (0x52ec3)
[2020-12-26 05:03:44.020 27967/F619] B[0003]: /usr/bin/pihole-FTL(match_regex+0x103) [0x56240d19c703]
[2020-12-26 05:03:44.021 27967/F619] L[0003]: N/A (0x4d703)
[2020-12-26 05:03:44.021 27967/F619] B[0004]: /usr/bin/pihole-FTL(+0x464c6) [0x56240d1954c6]
[2020-12-26 05:03:44.022 27967/F619] L[0004]: N/A (0x464c6)
[2020-12-26 05:03:44.022 27967/F619] B[0005]: /usr/bin/pihole-FTL(_FTL_new_query+0x76a) [0x56240d19694a]
[2020-12-26 05:03:44.023 27967/F619] L[0005]: N/A (0x4794a)
[2020-12-26 05:03:44.023 27967/F619] B[0006]: /usr/bin/pihole-FTL(tcp_request+0x804) [0x56240d1cf9b4]
[2020-12-26 05:03:44.023 27967/F619] L[0006]: N/A (0x809b4)
[2020-12-26 05:03:44.023 27967/F619] B[0007]: /usr/bin/pihole-FTL(+0x76164) [0x56240d1c5164]
[2020-12-26 05:03:44.024 27967/F619] L[0007]: N/A (0x76164)
[2020-12-26 05:03:44.024 27967/F619] B[0008]: /usr/bin/pihole-FTL(main_dnsmasq+0x127c) [0x56240d1c710c]
[2020-12-26 05:03:44.025 27967/F619] L[0008]: N/A (0x7810c)
[2020-12-26 05:03:44.025 27967/F619] B[0009]: /usr/bin/pihole-FTL(main+0xf1) [0x56240d18f301]
[2020-12-26 05:03:44.026 27967/F619] L[0009]: N/A (0x40301)
[2020-12-26 05:03:44.026 27967/F619] B[0010]: /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xeb) [0x7f1b407e009b]
[2020-12-26 05:03:44.026 27967/F619] B[0011]: /usr/bin/pihole-FTL(_start+0x2a) [0x56240d18f4da]
[2020-12-26 05:03:44.027 27967/F619] L[0011]: N/A (0x404da)
[2020-12-26 05:03:44.027 27967/F619] ------ Listing content of directory /dev/shm ------
[2020-12-26 05:03:44.027 27967/F619] File Mode User:Group  Filesize Filename
[2020-12-26 05:03:44.027 27967/F619] rwxrwxrwx root:root 260 .
[2020-12-26 05:03:44.027 27967/F619] rwxr-xr-x root:root 3K ..
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 4K FTL-per-client-regex
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 336K FTL-dns-cache
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 45K FTL-overTime
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 1M FTL-queries
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 20K FTL-upstreams
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 176K FTL-clients
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 98K FTL-domains
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 123K FTL-strings
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 12 FTL-settings
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 152 FTL-counters
[2020-12-26 05:03:44.027 27967/F619] rw------- pihole:pihole 48 FTL-lock
[2020-12-26 05:03:44.027 27967/F619] ---------------------------------------------------
[2020-12-26 05:03:44.027 27967/F619] Please also include some lines from above the !!!!!!!!! header.
```

## Debug Token

https://tricorder.pi-hole.net/kfnbdbo3to

**Comment by smx97** [comment]
Could you try to update again? From your log (9th line) it appears that you're still using the old FTL version 5.3.2 instead of the 5.3.4.

**Comment by libertyx82** [comment]
i have updated pihole this morning after the crash.

**Comment by DL6ER** [comment]
@libertyx82 Thanks for your report. The crash you reported was a known issue in v5.3.2 and got fixed in v5.3.3. Hence, I'll close this ticket as the exact same crash shouldn't happen again. Wish all a good time!
